### PR TITLE
bugfix: stabilize e2e waits and harden budget test setup

### PR DESCRIPTION
**Pull request by iguliaev**
## Why
E2E coverage on this branch included flaky failures around UI synchronization and intermittent test-user setup failures in budget tests. This change set improves stability so CI runs are deterministic and setup/teardown failures do not mask root errors.

## What Changed
- Files or areas updated:
  - `apps/web-next/e2e/tests/transactions.spec.ts`
  - `apps/web-next/e2e/tests/dashboard-month-range.spec.ts`
  - `apps/web-next/e2e/tests/budgets.spec.ts`
  - `apps/web-next/e2e/utils/test-helpers.ts`
  - Existing branch changes to bulk upload/data reset/settings resilience tests and plan docs are also included in this branch history.
- New functionality added:
  - Reusable helper utilities for deterministic readiness checks and visible Ant Design dropdown selection in e2e tests.
  - Retry/backoff for transient Supabase Auth `createUser` failures in e2e setup.
- Existing behavior changed:
  - Replaced brittle `networkidle` synchronization in affected tests with explicit UI-state assertions.
  - Budget test teardown/setup is now guarded when `beforeAll` fails to avoid cascade errors.

## Key Decisions
- Decision:
  - Use condition-based waits and scoped dropdown locators instead of page-level network-idle waits.
- Reasoning:
  - `networkidle` was flaky for app behavior with background requests and async UI transitions.
- Alternatives considered:
  - Increasing timeouts/retries only; rejected as it treats symptoms, not synchronization root cause.

## How This Affects the System
- User-facing changes:
  - None (test-only changes).
- API changes:
  - None.
- Performance implications:
  - None in production.
- Database changes:
  - None.
- Breaking changes:
  - None.

## Testing
- New tests added:
  - No brand-new spec files; existing e2e tests were refactored/hardened.
- Existing tests status:
  - `npm run test:e2e:ci -- e2e/tests/budgets.spec.ts` passed.
  - `npm run test:e2e:ci` passed (80/80).
- Manual testing performed:
  - N/A (automated e2e focus).
- Edge cases tested:
  - Transient auth setup failures and setup-failure teardown behavior in budget specs.
  - Dropdown interaction stability and chart/edit form readiness timing.
- Test files modified or added:
  - `apps/web-next/e2e/tests/transactions.spec.ts`
  - `apps/web-next/e2e/tests/dashboard-month-range.spec.ts`
  - `apps/web-next/e2e/tests/budgets.spec.ts`
  - `apps/web-next/e2e/utils/test-helpers.ts`

### CODE REVIEW

File: apps/web-next/e2e/utils/test-helpers.ts [hunk 1]
```diff
@@ -122,6 +140,30 @@ export async function waitForFormReady(page: Page, testId: string) {
...
+export async function selectFromVisibleAntdDropdown(
+  page: Page,
+  comboboxName: string,
+  optionTitle: string
+) {
+  await page.getByRole("combobox", { name: comboboxName }).click({ force: true });
+  const option = page
+    .locator(".ant-select-dropdown:visible")
+    .getByTitle(new RegExp(`^${optionTitle}$`, "i"));
+  await option.waitFor({ state: "visible" });
```
[the remark this answers is not in the document]
Comment (iguliaev) [reply apps/web-next/e2e/utils/test-helpers.ts:164]: Fixed in `apps/web-next/e2e/utils/test-helpers.ts`: `selectFromVisibleAntdDropdown()` now escapes regex metacharacters in `optionTitle` before building the anchored RegExp, so option labels like `A+B` or `(test)` are matched literally.
